feat: 소셜 로그인 및 회원 가입, 토큰 재발급

m1njae opened the pull request:
<!-- 
- PR 제목: [feat/fix/refactor...] 작업 내용 한 줄 요약 (브랜치 이름)
- commit message가 적절한지 확인해주세요.
- 적절한 branch로 요청했는지 확인해주세요.
- Assignees, Label을 붙여주세요.
- 주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 Reviewer로 등록해주세요.
- PR이 승인된 경우 해당 브랜치는 삭제 부탁드립니다. 
-->


## 📝 PR 요약

<!--
해당 pr에서 작업한 내역을 적어주세요.
-->

- 소셜 로그인 및 회원 가입, 토큰 재발급을 진행합니다

<br/>

## 📌 변경 사항 및 주의 사항

<!--
변경사항 및 주의 사항이 있다면 적어주세요.
주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 리뷰어로 등록해주세요. (다른 사람이 작성한 코드 수정 등)
코드 리뷰 시 더 꼼꼼하게 확인 받고 싶은 부분이 있다면 적어주세요.
-->
- 구글 accessToken을 받아 사용자 정보를 조회하는데  회원번호를 integer로 받으려고 했으나 회원번호가 제법 길어서 integer의 범위를 넘어서더라구요..? 그래서 급한대로 type을 string으로 옮겨놓았습니다. ~string으로 가져갈지 아니면 bigint와 같은 값으로 숫자로 가져갈 지 이 부분은 체다나랑 논의해보고 싶어요!~
https://stackoverflow.com/questions/35359385/postgresql-storing-a-value-as-integer-vs-varchar
bigint로 가져가겠습니다~
![image](https://user-images.githubusercontent.com/82744423/214504731-ed94a2e8-ad9a-4530-b429-689c6b66f4d7.png)
위 글은 userId type string에서 bigint로 바꾼 코드 보실 때 참고해주세요!


<br/>

![image](https://user-images.githubusercontent.com/82744423/214238741-d1714b5a-6543-4342-b0b7-27184a3f1029.png)

![image](https://user-images.githubusercontent.com/82744423/214238782-976aa673-c4f0-4587-b68d-ae4a83e8b17d.png)

![image](https://user-images.githubusercontent.com/82744423/214238808-7027673e-3b23-4de5-95fb-4979c0b9f03f.png)




## 🔗 Linked Issue
close #15

m1njae committed:
feat: 소셜 로그인 및 회원 가입 API

토큰을 받지 않아 전반적인 로직만 작성해놓은 상태입니다. 토큰을 받아 API 테스트하면서 발생하는 오류들을 수정하겠습니다.

m1njae committed:
feat: 토큰 재발급

m1njae committed:
chore: entities 수정 적용

m1njae committed:
chore: userId type 변경

구글 회원번호가 integer 범위를 넘어가 급한대로  type을 string으로 변경해놓았습니다.

m1njae committed:
fix: 소셜 로그인 및 회원가입, 토큰 재발급 로직 수정

m1njae committed:
chore: userId type string에서 bigint로 변경

m1njae committed:
chore: transaction 삭제 및 save 메소드 insert로 변경

Chedda98 commented on src/controllers/AuthController.ts:
```diff
@@ -0,0 +1,114 @@
+          .status(statusCode.UNAUTHORIZED)
+          .send(fail(statusCode.UNAUTHORIZED, message.INVALID_TOKEN));
+      }
+      if (user == exceptionMessage.INVALID_USER) {
```
@m1njae 여기서 궁금한 점이 하나 있슴니다!!
처음 가입하는 로직에서 user가 없는 것이랑 유효하지 않은 유저랑 차이가 
`!user`: 드래그미에 처음 가입하는 유저? -> 라면 왜 "유효하지 않은 토큰"이라는 메시지를 출력하는지도 궁금합니다!! 
`INVALID_USER`: 토큰 만료로 재로그인이 필요한 유저
요런 차이일까요?!!

m1njae replied:
@Chedda98  `!user` 는 구글 사용자 정보에 대한 값이 없을 경우입니다! 구글 accessToken이 유효하지 않을 경우에는 사용자 정보가 출력이 되지 않을거라고 판단했습니다.
`INVALID_USER`는 구글 accessToken을 받아서 userId가 없을 때, INVALID_USER를 반환하도록 작성했습니다!
근데 생각해보니, 두 코드가 모두 같은 에러처리를 하는 것 같아서 수정해놓도록 하겠습니다 고마워요 체다~

Chedda98 commented on src/services/UserService.ts:
```diff
@@ -24,19 +23,17 @@ export class UserService {
+        email: createUserDto.email,
+        nick: createUserDto.nick,
+        provider: createUserDto.provider,
+        snsId: createUserDto.snsId,
```
👍🏻

m1njae replied:
지금 코드 다시 보는데 오류가 생긴 걸 확인해서 수정해놓도록 하겠읍니다..!

m1njae committed:
chore: insert 메소드 save으로 변경

insert로 변경 시, 반환타입이 InsertResult가 되어 로직에 문제가 생깁니다.
단순히 테이블에 값을 추가하는 경우에는 insert 메소드를 사용하되,
생성한 값을 가지고 로직을 이어나가야 하는 경우에는 save 메소드를 사용하도록 하겠습니다.

m1njae committed:
chore: 코드리뷰 피드백 반영

dannaward reviewed (approved):
짱짱맨!!!

m1njae committed:
chore: refreshtoken 만료주기 30일로 변경
